File: RayTracerModel/Intersection.cs
namespace RayTracer
{
    public class Intersection
    {
        public Thing Thing { get; private set; }
        public Ray Ray { get; private set; }
        public double Dist { get; private set; }

        public Intersection(Thing thing, Ray ray, double dist)
        {
            Thing = thing;
            Ray = ray;
            Dist = dist;
        }
    }
}

File: RayTracerVBSingleProj/Intersection.vb
Public Class Intersection

    Public ReadOnly Property Thing As Thing
    Public ReadOnly Property Ray As Ray
    Public ReadOnly Property Dist As Double

    Public Sub New(ByVal thing As Thing, ByVal ray As Ray, ByVal dist As Double)
        Me.Thing = thing
        Me.Ray = ray
        Me.Dist = dist
    End Sub
End Class
